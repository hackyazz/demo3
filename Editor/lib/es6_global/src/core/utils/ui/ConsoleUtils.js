

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as LogUtils$WonderEditor from "../console/LogUtils.js";

function _console(message, editorState, param) {
  return Curry._1(param[1], message);
}

function warn(message, editorState) {
  return _console(message, editorState, /* tuple */[
              (function (messageObj) {
                  return messageObj.warn(message, 4);
                }),
              LogUtils$WonderEditor.warn
            ]);
}

function log(message, editorState) {
  return _console(message, editorState, /* tuple */[
              (function (messageObj) {
                  return messageObj.log(message, 4);
                }),
              LogUtils$WonderEditor.log
            ]);
}

function info(message, editorState) {
  return _console(message, editorState, /* tuple */[
              (function (messageObj) {
                  return messageObj.info(message, 4);
                }),
              LogUtils$WonderEditor.info
            ]);
}

function debug(buildMessageFunc, isDebug, editorState) {
  return LogUtils$WonderEditor.debug(buildMessageFunc, isDebug);
}

function error(message, editorState) {
  return _console(message, editorState, /* tuple */[
              (function (messageObj) {
                  return messageObj.error(message, 4);
                }),
              LogUtils$WonderEditor.error
            ]);
}

var logStack = LogUtils$WonderEditor.logStr;

export {
  _console ,
  warn ,
  log ,
  info ,
  debug ,
  error ,
  logStack ,
  
}
/* LogUtils-WonderEditor Not a pure module */
