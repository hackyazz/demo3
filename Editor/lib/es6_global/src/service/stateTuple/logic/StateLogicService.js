

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as StateEditorService$WonderEditor from "../../state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../state/engine/state/StateEngineService.js";
import * as DirectorEngineService$WonderEditor from "../../state/engine/DirectorEngineService.js";

function getEngineStateToGetData(handleFunc) {
  return Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0));
}

function getAndSetEngineState(handleFunc) {
  StateEngineService$WonderEditor.setState(Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function refreshEngineState(engineState) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, engineState));
  return /* () */0;
}

function refreshEngineStateAndReturnEngineState(engineState) {
  return DirectorEngineService$WonderEditor.loopBody(0, engineState);
}

function getAndRefreshEngineState(param) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function getAndRefreshEngineStateWithFunc(handleFunc) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0))));
  return /* () */0;
}

function loopBodyWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getAndRefreshEngineStateWhenStop(param) {
  StateEngineService$WonderEditor.setState(loopBodyWhenStop(StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function renderWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getEditorState(handleFunc) {
  return Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0));
}

function getAndSetEditorState(handleFunc) {
  StateEditorService$WonderEditor.setState(Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0)));
  return /* () */0;
}

function getStateToGetData(handleFunc) {
  return Curry._1(handleFunc, /* tuple */[
              StateEditorService$WonderEditor.getState(/* () */0),
              StateEngineService$WonderEditor.unsafeGetState(/* () */0)
            ]);
}

function getAndSetState(handleFunc) {
  var match = Curry._1(handleFunc, /* tuple */[
        StateEditorService$WonderEditor.getState(/* () */0),
        StateEngineService$WonderEditor.unsafeGetState(/* () */0)
      ]);
  StateEditorService$WonderEditor.setState(match[0]);
  StateEngineService$WonderEditor.setState(match[1]);
  return /* () */0;
}

function setState(param) {
  StateEditorService$WonderEditor.setState(param[0]);
  StateEngineService$WonderEditor.setState(param[1]);
  return /* () */0;
}

export {
  getEngineStateToGetData ,
  getAndSetEngineState ,
  refreshEngineState ,
  refreshEngineStateAndReturnEngineState ,
  getAndRefreshEngineState ,
  getAndRefreshEngineStateWithFunc ,
  loopBodyWhenStop ,
  getAndRefreshEngineStateWhenStop ,
  renderWhenStop ,
  getEditorState ,
  getAndSetEditorState ,
  getStateToGetData ,
  getAndSetState ,
  setState ,
  
}
/* StateEditorService-WonderEditor Not a pure module */
