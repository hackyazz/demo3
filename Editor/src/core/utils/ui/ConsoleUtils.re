let _console = (message, editorState, (messageConsoleFunc, logConsoleFunc)) => {
  DebugSettingEditorService.isNotShowMessage(editorState) ?
    () :
    editorState
    |> messageConsoleFunc(message, 4000)
    |> StateEditorService.setState
    |> ignore;

  logConsoleFunc(message);
};

let warn = (message, editorState) =>
  _console(
    message,
    editorState,
    (MessageUIEditorService.warn, LogUtils.warn),
  );

let log = (message, editorState) =>
  _console(message, editorState, (MessageUIEditorService.log, LogUtils.log));

let info = (message, editorState) =>
  _console(
    message,
    editorState,
    (MessageUIEditorService.info, LogUtils.info),
  );

let debug = (buildMessageFunc, isDebug, editorState) =>
  /* DebugSettingEditorService.isNotShowMessage(editorState) ?
     () :
     Antd.Message.message
     |> Antd.Message.convertToJsObj
     |> (messageObj => messageObj##info(buildMessageFunc(), 4))
     |> ignore; */
  LogUtils.debug(buildMessageFunc, isDebug);

let error = (message, editorState) =>
  _console(
    message,
    editorState,
    (MessageUIEditorService.error, LogUtils.error),
  );

let logStack = stack => LogUtils.logStr(stack);