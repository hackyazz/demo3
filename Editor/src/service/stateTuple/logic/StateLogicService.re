let getEngineStateToGetData = handleFunc =>
  StateEngineService.unsafeGetState() |> handleFunc;

let getAndSetEngineState = handleFunc =>
  StateEngineService.unsafeGetState()
  |> handleFunc
  |> StateEngineService.setState
  |> ignore;

let refreshEngineState = engineState =>
  engineState
  |> DirectorEngineService.loopBody(0.)
  |> StateEngineService.setState
  |> ignore;

let refreshEngineStateAndReturnEngineState = engineState =>
  engineState |> DirectorEngineService.loopBody(0.);

let getAndRefreshEngineState = () =>
  StateEngineService.unsafeGetState()
  |> DirectorEngineService.loopBody(0.)
  |> StateEngineService.setState
  |> ignore;

let getAndRefreshEngineStateWithFunc = handleFunc =>
  StateEngineService.unsafeGetState()
  |> handleFunc
  |> DirectorEngineService.loopBody(0.)
  |> StateEngineService.setState
  |> ignore;

let loopBodyWhenStop = engineState =>
  StateEditorService.getIsRun() ?
    engineState : engineState |> DirectorEngineService.loopBody(0.);

let getAndRefreshEngineStateWhenStop = () =>
  StateEngineService.unsafeGetState()
  |> loopBodyWhenStop
  |> StateEngineService.setState
  |> ignore;

let renderWhenStop = engineState =>
  StateEditorService.getIsRun() ?
    engineState : engineState |> DirectorEngineService.loopBody(0.);

let getEditorState = handleFunc => StateEditorService.getState() |> handleFunc;

let getAndSetEditorState = handleFunc =>
  StateEditorService.getState()
  |> handleFunc
  |> StateEditorService.setState
  |> ignore;

let getStateToGetData = handleFunc =>
  (StateEditorService.getState(), StateEngineService.unsafeGetState())
  |> handleFunc;

let getAndSetState = handleFunc => {
  let (editorState, engineState) =
    (StateEditorService.getState(), StateEngineService.unsafeGetState())
    |> handleFunc;

  editorState |> StateEditorService.setState |> ignore;
  engineState |> StateEngineService.setState |> ignore;

  ();
};

let setState = ((editorState, engineState)) => {
  editorState |> StateEditorService.setState |> ignore;
  engineState |> StateEngineService.setState |> ignore;

  ();
};