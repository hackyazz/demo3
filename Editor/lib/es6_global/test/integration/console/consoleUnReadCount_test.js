

import * as Curry from "../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as TestTool$WonderEditor from "../../tool/TestTool.js";
import * as ConsoleTool$WonderEditor from "../../unit/tool/external/ConsoleTool.js";
import * as ConsoleUtils$WonderEditor from "../../../src/core/utils/ui/ConsoleUtils.js";
import * as ReactTestTool$WonderEditor from "../../tool/ReactTestTool.js";
import * as ConsoleStoreTool$WonderEditor from "./tool/ConsoleStoreTool.js";
import * as StateLogicService$WonderEditor from "../../../src/service/stateTuple/logic/StateLogicService.js";
import * as BuildComponentTool$WonderEditor from "../../tool/BuildComponentTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../tool/MainEditorSceneTool.js";

describe("test console unread count", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                MainEditorSceneTool$WonderEditor.prepareScene(sandbox);
                return ConsoleTool$WonderEditor.notShowMessage(/* () */0);
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("should show unread count in bottom header", (function () {
                beforeEach((function () {
                        BuildComponentTool$WonderEditor.buildBottom(undefined, undefined, /* () */0);
                        return /* () */0;
                      }));
                Wonder_jest.test("if console one message, unread count should + 1", (function () {
                        ConsoleStoreTool$WonderEditor.buildStore(/* Project */0, /* () */0);
                        var partial_arg = "warn message";
                        StateLogicService$WonderEditor.getEditorState((function (param) {
                                return ConsoleUtils$WonderEditor.warn(partial_arg, param);
                              }));
                        return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildBottomHeader(TestTool$WonderEditor.buildEmptyAppState(/* () */0), undefined, /* () */0));
                      }));
                return Wonder_jest.test("\n        1.warn\n        2.show Console ui\n\n        unread count should be clear to 0\n        ", (function () {
                              var store = ConsoleStoreTool$WonderEditor.buildStore(/* Console */1, /* () */0);
                              var partial_arg = "warn message";
                              StateLogicService$WonderEditor.getEditorState((function (param) {
                                      return ConsoleUtils$WonderEditor.warn(partial_arg, param);
                                    }));
                              BuildComponentTool$WonderEditor.buildConsole(store, undefined, /* () */0);
                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildBottomHeader(store, undefined, /* () */0));
                            }));
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
