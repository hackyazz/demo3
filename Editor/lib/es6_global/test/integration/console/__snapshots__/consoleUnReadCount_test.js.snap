// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test console unread count should show unread count in bottom header 
        1.warn
        2.show Console ui

        unread count should be clear to 0
         1`] = `
<article
  className="bottom-header"
>
  <div
    className="bottom-widget-category"
  >
    <div
      className="category-name"
      onClick={[Function]}
    >
      <div
        className="name-header"
      >
        Project
      </div>
    </div>
    <div
      className="category-name category-active"
      onClick={[Function]}
    >
      <div
        className="name-header"
      >
        Console
      </div>
    </div>
    <span
      className="category-name"
    />
  </div>
</article>
`;

exports[`test console unread count should show unread count in bottom header if console one message, unread count should + 1 1`] = `
<article
  className="bottom-header"
>
  <div
    className="bottom-widget-category"
  >
    <div
      className="category-name category-active"
      onClick={[Function]}
    >
      <div
        className="name-header"
      >
        Project
      </div>
    </div>
    <div
      className="category-name"
      onClick={[Function]}
    >
      <div
        className="name-header"
      >
        Console
      </div>
      <div
        className="name-tail"
      >
        1
      </div>
    </div>
    <span
      className="category-name"
    />
  </div>
</article>
`;
